tools/python/boutiques/localExec.py

```diff
@@ -1047,6 +1047,68 @@ def _generateOutputFileNames(self):
                 outputFileName = os.path.abspath(outputFileName)
             self.out_dict[outputId] = outputFileName
 
+        # if 'conditional-path-template' in outputItem
+        # (key=conditions, value=path)
+        # Initialize file name with path template or existing value
+        for outputId in [x['id'] for x in self.outputs
+                         if 'conditional-path-template' in x]:
+
+            for boolObj in self.safeGet(outputId, 'conditional-path-template'):
+                # Surround expression chars with space
+                # Then split by space to isolate ids
+                boolExp = list(boolObj)[0]
+                splitExp = "".join(
+                    [c if c.isalnum() or c == "_" else
+                     " {0} ".format(c) for c in boolExp]).split(" ")
+                parsedExp = []
+                for word in [word.strip() for word in splitExp]:
+                    all_ids = [i['id'] for i in (self.inputs + self.outputs)]
+                    # Substitute boolean expression key by its value
+                    in_out_dict = self.in_dict.copy()
+                    in_out_dict.update(self.out_dict)
+                    if word in in_out_dict:
+                        value = "{0}".format(in_out_dict[word])
+                        if value.replace(".", "").replace("-", "").isdigit():
+                            parsedExp.append(in_out_dict[word])
+                        else:
+                            parsedExp.append("\"{0}\"".format(value))
+                    # Boolean expression key is not chosen (optional input),
+                    # therefore expression is false
+                    elif word in all_ids:
+                        parsedExp = ["False"]
+                        break
+                    # Word is an expression char, just append it
+                    else:
+                        parsedExp.append(word)
+                # If expression is true, set fileName
+                # Stop checking (if-elif...)
+                if " ".join("{0}".format(w) for w in parsedExp) == "default":
```

Comment:
I may be reading this wrong, but aren't what you have written on #L1078 (`parsedExp = ["False"]`) and what you're checking here, (` .... == "default"`) incompatible, so defaults would never be added? You know what's going on in here much better than I, @DarrinFong, so please just double check and let me know :)

Reply:
```python3
if word in in_out_dict:
    # in_out_dict are k:v pairs that will be in the output of simulate
elif word in all_ids:
    parsedExp = ["False"]
    break
```
The elif block should only 'falsify' the expressions that contain an ID which is both:
- in descriptor's inputs/output-files IDs
- NOT in selected values for simulate
Which means the only way for ```'default'``` to be evaluated ```False``` is when an ID of name ```'default'``` is present in the descriptor's inputs/output-files IDs. (This raises a concern of whether we have to add ```'default'``` to the list of invalid IDs).

Reply:
Great - thank you for the clarification :)

I do think that maybe we want to add "default" as a restricted word in the schema so that it can't be used as an ID... @glatard, thoughts?